mlir/lib/Dialect/GPU/Transforms/ParallelLoopMapper.cpp

```diff
@@ -65,16 +66,29 @@ static MappingLevel &operator++(MappingLevel &mappingLevel) {
   return mappingLevel;
 }
 
+// Map the policy string to a typed mapping policy.
+static FailureOr<MappingPolicy> getMappingPolicyFromStr(StringRef policy) {
+  std::string policyCanonical = policy.trim().lower();
+
+  auto option =
+      llvm::StringSwitch<std::optional<MappingPolicy>>(policyCanonical)
```

Comment:
Don't use auto.

Reply:
Type is visible on the right-hand side. So I presume the use of `auto` here is fine?